File: backend/src/main/java/com/team01/backend/domain/category/entity/Category.java
package com.team01.backend.domain.category.entity;

import com.team01.backend.global.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@Getter
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "boardId", "name" }) })
public class Category extends BaseEntity {
    private Long boardId;
    private String name;

    public Category(Long boardId, String name){
        this.boardId = boardId;
        this.name = name;
    }

    public void update(String name) {
        this.name = name;
    }
}

File: backend/src/main/java/com/team01/backend/domain/category/entity/Category.kt
package com.team01.backend.domain.category.entity

import com.team01.backend.global.entity.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import lombok.Getter

@Entity
@Getter
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["boardId", "name"])])
class Category(boardId: Long, name: String) : BaseEntity() {
    var boardId: Long = boardId
        protected set

    var name: String = name
        protected set

    fun update(name: String) {
        this.name = name
    }
}